src/main/java/frc/robot/subsystems/swerve/swervecontainer/SwerveFactory.java

```diff
@@ -2,47 +2,55 @@
 
 import frc.robot.subsystems.swerve.SwerveConstants;
 import frc.robot.subsystems.swerve.gyro.ISwerveGyro;
-import frc.robot.subsystems.swerve.gyro.pigeon2.Pigeon2Gyro;
 import frc.robot.subsystems.swerve.gyro.replay.ReplaySwerveGyro;
-import frc.robot.subsystems.swerve.gyro.simulation.SimulationSwerveGyro;
 import frc.robot.subsystems.swerve.modules.Module;
 import frc.robot.subsystems.swerve.modules.ModuleUtils;
 import frc.robot.subsystems.swerve.modules.replay.ReplayModule;
-import frc.robot.subsystems.swerve.modules.simulation.SimulationModule;
-import frc.robot.subsystems.swerve.modules.talonfx.TalonFXModule;
+import frc.robot.subsystems.swerve.swervecontainer.real.RealSwerve;
+import frc.robot.subsystems.swerve.swervecontainer.simulation.SimulationSwerve;
 
 import static frc.robot.Robot.ROBOT_TYPE;
 
 public class SwerveFactory {
 
+    public static class SwerveContainerKey {
+
+        private SwerveContainerKey(){}
+
+    }
+
+    private static final SwerveContainer REAL_SWERVE = new RealSwerve(new SwerveContainerKey());
+    private static final SwerveContainer SIMULATION_SWERVE = new SimulationSwerve(new SwerveContainerKey());
+
+
     public static SwerveConstants createConstants() {
         return switch (ROBOT_TYPE) {
-            case REAL, REPLAY -> RealSwerveContainer.swerveConstants;
-            case SIMULATION -> SimulationSwerveContainer.swerveConstants;
+            case REAL, REPLAY -> REAL_SWERVE.getConstants();
+            case SIMULATION -> SIMULATION_SWERVE.getConstants();
         };
     }
 
     public static ISwerveGyro createGyro() {
         return switch (ROBOT_TYPE) {
-            case REAL -> new Pigeon2Gyro(RealSwerveContainer.pigeon2GyroConfigObject);
-            case SIMULATION -> new SimulationSwerveGyro();
+            case REAL -> REAL_SWERVE.getGyro();
+            case SIMULATION -> SIMULATION_SWERVE.getGyro();
             case REPLAY -> new ReplaySwerveGyro();
         };
     }
 
     public static Module[] createModules(){
         return switch (ROBOT_TYPE) {
             case REAL -> new Module[]{
-                    new Module(ModuleUtils.ModuleName.FRONT_LEFT, new TalonFXModule(RealSwerveContainer.moduleConstants[0])),
-                    new Module(ModuleUtils.ModuleName.FRONT_RIGHT, new TalonFXModule(RealSwerveContainer.moduleConstants[1])),
-                    new Module(ModuleUtils.ModuleName.BACK_LEFT, new TalonFXModule(RealSwerveContainer.moduleConstants[2])),
-                    new Module(ModuleUtils.ModuleName.BACK_RIGHT, new TalonFXModule(RealSwerveContainer.moduleConstants[3]))
+                    new Module(ModuleUtils.ModuleName.FRONT_LEFT, REAL_SWERVE.getModules()[0]),
+                    new Module(ModuleUtils.ModuleName.FRONT_RIGHT, REAL_SWERVE.getModules()[1]),
+                    new Module(ModuleUtils.ModuleName.BACK_LEFT, REAL_SWERVE.getModules()[2]),
+                    new Module(ModuleUtils.ModuleName.BACK_RIGHT, REAL_SWERVE.getModules()[3])
```

Comment:
make a function that takes in a SwerveContainer and returns all new modules instead of doing it this way